flink-table/flink-table-common/src/test/java/org/apache/flink/table/functions/UserDefinedFunctionHelperTest.java

```diff
@@ -143,34 +162,27 @@ private void testErrorMessage() {
 
         @Nullable String expectedErrorMessage;
 
-        TestSpec(Class<? extends UserDefinedFunction> functionClass) {
+        TestSpec(
+                @Nullable Class<? extends UserDefinedFunction> functionClass,
+                @Nullable UserDefinedFunction functionInstance,
+                @Nullable CatalogFunction catalogFunction) {
             this.functionClass = functionClass;
-            this.functionInstance = null;
-            this.catalogFunction = null;
-        }
-
-        TestSpec(UserDefinedFunction functionInstance) {
-            this.functionClass = null;
             this.functionInstance = functionInstance;
-            this.catalogFunction = null;
-        }
-
-        TestSpec(CatalogFunction catalogFunction) {
-            this.functionClass = null;
-            this.functionInstance = null;
             this.catalogFunction = catalogFunction;
         }
 
         static TestSpec forClass(Class<? extends UserDefinedFunction> function) {
-            return new TestSpec(function);
+            return new TestSpec(function, null, null);
         }
 
         static TestSpec forInstance(UserDefinedFunction function) {
-            return new TestSpec(function);
+            return new TestSpec(null, function, null);
         }
 
-        static TestSpec forCatalogFunction(String className, FunctionLanguage language) {
-            return new TestSpec(new CatalogFunctionMock(className, language));
+        static TestSpec forCatalogFunction(
+                String className,
+                @SuppressWarnings("SameParameterValue") FunctionLanguage language) {
```

Comment:
What warning is this one?

Reply:
You are right. I will simply always choose Java for now.